api/server/router/container/container_routes.go

```diff
@@ -576,9 +576,15 @@ func (s *containerRouter) wsContainersAttach(ctx context.Context, w http.Respons
 		UseStdin:   true,
 		UseStdout:  true,
 		UseStderr:  true,
-		MuxStreams: false, // TODO: this should be true since it's a single stream for both stdout and stderr
+		MuxStreams: true,
```

Comment:
I'm curious about that - [the docs](https://docs.docker.com/engine/api/v1.32/#operation/ContainerAttachWebsocket) include options like `stdout` and `stderr` to disable separate streams .. how does the muxing work with all this?

i'd think that we either would need to use the logic as we already have in `postContainersAttach` or at least to update the docs for the websockets .. is it just me not getting the picture?

Reply:
Currently the streams are not muxed so there is no way to differentiate stdout and stderr on the stream.
Technically the client could just ask for two streams.

Reply:
thanks for getting back @cpuguy83 

that is what i was trying to say and ask somewhat at the same time. since both options `stdin` and `stdout` are always going to be `true` no matter what .. how would you see the client do that? the docs mention options for flipping specific streams on and off - but given the actual code (w/ as well as w/o your change) it doesn't seem to offer that .. does it?

Reply:
You are right, it's always true... This endpoint was never designed correctly.

Reply:
so any idea where we're going with this going forward? for the time being i can split the messages i'm getting on same pattern, but nevertheless i'd very much like to see separated streams even on the websockets (: